src/pubsub.ts

```diff
@@ -182,26 +182,29 @@ export class SubscriptionManager {
             // convention this is the value returned from the mutation
             // resolver
             const onMessage = (rootValue) => {
-                let contextPromise;
-                if (typeof options.context === 'function') {
-                    contextPromise = new Promise((resolve) => {
-                        resolve(options.context());
-                    });
-                } else {
-                    contextPromise = Promise.resolve(options.context);
-                }
-                contextPromise.then((context) => {
-                    if (!filter(rootValue, context)) {
+                Promise.resolve().then(() => {
```

Comment:
@NeoPhi , so we can resolve it any way, and do the same for context, and then use `Promise.all`... something like that: 

```ts
const onMessage = (rootValue) => {
    const contextResult = typeof options.context === 'function' ? options.context() : context;

    return Promise
        .resolve(contextResult)
        .then(context => Promise.all([ Promise.resolve(filter(rootValue, context)), Promise.resolve(context) ]) )
        .then(promisesResults=> {
            const filterResult = promisesResults[0];
            const context = promisesResults[1];

            if (!filterResult) return;

            return execute(
                this.schema,
                parsedQuery,
                rootValue,
                context,
                options.variables,
                options.operationName
            );
        })
        .then(executionResult => {
            options.callback(null, executionResult);
        })
        .catch((error) => {
            options.callback(error);
        });
}
```